runtime/gc_base/StringTable.cpp

```diff
@@ -623,6 +623,7 @@ j9gc_createJavaLangString(J9VMThread *vmThread, U_8 *data, UDATA length, UDATA s
 				while (tempLength != 0) {
 					U_16 unicode = 0;
 					UDATA consumed = VM_VMHelpers::decodeUTF8CharN(tempData, &unicode, tempLength);
+					Assert_MM_true(0 != consumed);
```

Comment:
I believe you can use assertion with message like `Assert_GC_true_with_message`, one of defined in https://github.com/eclipse/omr/blob/master/gc/base/ModronAssertions.h
It provides similar handling as for call in VM https://github.com/eclipse/openj9/blob/868f0ec9bd1a3e10f284f02c1f82703bfb020a29/runtime/vm/jnimisc.cpp#L253 and allows to output more details

Reply:
Sure, improved with `Assert_GC_true_with_message2(MM_EnvironmentBase::getEnvironment(vmThread->omrVMThread), 0 != consumed, "Invalid UTF8 character at %p, %lu", tempData, tempLength)`.